distributed/tests/test_worker.py

```diff
@@ -1248,11 +1248,65 @@ def __reduce__(self):
     await wait(f1)
     assert set(a.data.memory) == {"f1"}
 
-    futures = c.map(OverReport, range(int(100e6), int(100e6) + 8))
+    with captured_logger("distributed.worker") as logger:
+        futures = c.map(OverReport, range(int(100e6), int(100e6) + 5))
+
+        while not a.data.disk:
+            await asyncio.sleep(0.01)
+        assert "f1" in a.data.disk
+        await asyncio.sleep(0.5)
+
+    # Spilling normally starts at the spill threshold and stops at the target threshold.
+    # In this special case, it stops as soon as the process memory goes below the spill
+    # threshold, e.g. without a hysteresis cycle. Test that we didn't instead dump the
+    # whole data to disk (memory_limit * target = 0)
+    assert "Unmanaged memory use is high" not in logger.getvalue()
```

Comment:
Not a great test, but I couldn't find anything better. Testing that a.fast is not empty is worthless as you end up in race conditions with some of the futures finishing after the spill starts running.